src/ui/components/AppWrapper/AppWrapper.tsx

```diff
@@ -251,19 +251,26 @@ const AppWrapper = (props: { children: ReactNode }) => {
   }, [authentication.loggedIn, initAppSuccess]);
 
   useEffect(() => {
-    PeerConnection.peerConnection.onPeerDisconnectedStateChanged(
-      async (event) => {
-        if (!connectedWallet) {
-          return;
-        }
-        return peerDisconnectedChangeHandler(
-          event,
-          connectedWallet.id,
-          dispatch
-        );
+    if (!connectedWallet?.id) {
+      return;
+    }
+
+    const eventHandler = async (event: PeerDisconnectedEvent) => {
+      if (!connectedWallet?.id) {
```

Comment:
We can actually remove this if statement now since you already did the check outside the event handler (which is a lot better, so it doesn't create an event handler that does nothing!)